### PR TITLE
Support for headless HITL apps, first pass

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

"Headless" means no attached local display. A headless mode is useful for (1) unit-testing on headless CI machines, and (2) the client/server model, with Pick_throw_vr running as a server (`remote_gui_mode`) on e.g. a headless cloud machine.

With this PR, the Unity client can connect to the headless server and e.g. view the animated scene. However, as a first pass, this doesn't actually 100% work with clients yet because of the way I've completely stubbed out GuiInput (which RemoteGuiInput needs).

## How Has This Been Tested

Locally tested on Macbook. Tested on headless Ubuntu EC2 instance with Unity client connected.

## Types of changes

- **\[HITL Framework\]** 

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/pick_throw_vr/config/experiment/headless_server.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# @package _global_
```
Comment (eundersander) [inline examples/hitl/pick_throw_vr/config/experiment/headless_server.yaml:1]: "experiment" is an odd name for a folder that conceptually contains config overrides, but Hydra is really confusing and I was looking to match the naming convention they use in their docs and examples. The name here is unrelated to my use of "experimental" in the config, btw--in the future, we'll put all config overrides in this folder, even if they're not experimental features.

**Comment by henrysamer** [comment]
Tested the changes from this PR on my Mac (Intel i9, Intel UHD Graphics 630).

**Review by henrysamer — APPROVED** [review]
-Docstrings for DebugVideoWriter and StubTextDrawer recommended.
-can keep the name Appdriver for now

Thanks!

**Review by 0mdc — APPROVED** [review]
LGTM!

**Pull request merged**